### ISSUE TITLE
apt_repository no longer updates apt cache on change

### ISSUE THREAD

**Issue by yumike** [ansible/ansible#4136]
In ansible 1.2.3 `apt_repository` automatically updated apt cache on change. As of 1.3.0 it doesn't.

```

---
- name: add postgresql apt key
  apt_key: url=http://apt.postgresql.org/pub/repos/apt/ACCC4CF8.asc state=present

- name: add postgresql apt repository
  apt_repository: repo='deb http://apt.postgresql.org/pub/repos/apt precise-pgdg main' state=present

- name: install postgresql server
  apt: name=postgresql-9.2 state=present
```

In this example ansible fails saying postgresql-9.2 doesn't found. However, after `apt-get update` postgresql-9.2 is available.

**Comment by resmo** [comment]
Change

```
apt: name=postgresql-9.2 state=present
```

to 

```
apt: name=postgresql-9.2 state=present update_cache=yes
```

**Comment by jimi-c** [comment]
As @resmo noted, you need to add that additional option. If you are still having issues with this, let us know and we can re-open this. Thanks!

**Comment by yumike** [comment]
Thanks for the workaround! But isn't this breaks backward compatibility? Also, am I understand correctly that

```
apt: name=postgresql-9.2 state=present update_cache=yes
```

always updates cache regardless of the fact that package is already present?

**Comment by srgvg** [comment]
@yumike yes, update_cache is a separate process from package installation, and will allways run (unless you set cache_valid_time)

**Comment by resmo** [comment]
this bug should be reopened.

updating cache makes totally sense if apt_repository changes. This functionality was lost in commit ded947008a2dff34c4389226f3fca5b926a0ccea

**Comment by marascio** [comment]
I agree. Not updating the cache after apt_repository seems wrong. Should re-open and merge the fix above.

**Comment by jimi-c** [comment]
I just merged @resmo's fix above, so no need to reopen this. Thanks!

**Comment by jimi-c** [comment]
@bryanlarsen this patch was included in 1.3.2.
